catroid/src/main/java/org/catrobat/catroid/ui/dialogs/FormulaEditorComputeDialog.java

```diff
@@ -145,22 +149,40 @@ public void onAccuracyChanged(Sensor sensor, int accuracy) {
 
 	@Override
 	public void onSensorChanged(SensorEvent event) {
-		showFormulaResult();
+		ProjectManager projectManager = ProjectManager.getInstance();
+		Sprite currentSprite = projectManager.getCurrentSprite();
+		AndroidStringProvider stringProvider = new AndroidStringProvider(context);
+		showFormulaResult(currentSprite, stringProvider);
 	}
 
 	private void setDialogTextView(final String newString) {
-		computeTextView.post(new Runnable() {
-			@Override
-			public void run() {
-				computeTextView.setText(newString);
-
-				ViewGroup.LayoutParams params = computeTextView.getLayoutParams();
-				int height = computeTextView.getLineCount() * computeTextView.getLineHeight();
-				int heightMargin = (int) (height * 0.5);
-				params.width = ViewGroup.LayoutParams.MATCH_PARENT;
-				params.height = height + heightMargin;
-				computeTextView.setLayoutParams(params);
-			}
+		computeTextView.post(() -> {
+			computeTextView.setText(newString);
+
+			ViewGroup.LayoutParams params = computeTextView.getLayoutParams();
+			int height = computeTextView.getLineCount() * computeTextView.getLineHeight();
+			int heightMargin = (int) (height * 0.5);
+			params.width = ViewGroup.LayoutParams.MATCH_PARENT;
+			params.height = height + heightMargin;
+			computeTextView.setLayoutParams(params);
 		});
 	}
+
+	private static class AndroidStringProvider implements Formula.StringProvider {
+		private final Context context;
```

Comment:
Patrick already mentioned this issue, please refactor to dont use context here for only fetching a string